pkg/synthetictests/allowedalerts/basic_alert.go

```diff
@@ -321,6 +398,20 @@ func (a *basicAlertTest) InvariantCheck(ctx context.Context, restConfig *rest.Co
 	)
 
 	state, message := a.failOrFlake(ctx, restConfig, firingIntervals, pendingIntervals)
+
+	switch a.alertName {
```

Comment:
I find this approach acceptable. With kube freeze coming I may not get back to this until next week.  Could you tidy up the import paths (depending on monitorapi is much better than monitor) and see if someone else can have a look as well.

Reply:
Sounds good -- thanks for the review!